app/controllers/publishers/ats_api/v1/vacancies_controller.rb

```diff
@@ -1,58 +1,79 @@
 class Publishers::AtsApi::V1::VacanciesController < Api::ApplicationController
   before_action :authenticate_client!
+  before_action :validate_payload, only: %i[create update]
+
+  rescue_from StandardError, with: :render_server_error
+  rescue_from ActiveRecord::RecordNotFound, with: :render_not_found
+  rescue_from ActionController::ParameterMissing, with: :render_bad_request
+  rescue_from Publishers::AtsApi::CreateVacancyService::InvalidOrganisationError, with: :render_bad_request
 
   def index
-    @pagy, @vacancies = pagy(vacancies, items: 100)
+    @pagy, @vacancies = pagy(vacancies.where(publisher_ats_api_client: client), items: 100)
 
     respond_to(&:json)
   end
 
   def show
-    @vacancy = vacancy
+    @vacancy = Vacancy.find(params[:id])
+
+    respond_to(&:json)
   end
 
-  # No idea why rubocop can't see the check after the create call
-  # rubocop:disable Rails/SaveBang
   def create
-    @vacancy = Vacancy.create(vacancy_params)
-
-    respond_to do |format|
-      format.json do
-        if @vacancy.persisted?
-          render status: :created
-        else
-          render status: :bad_request
-        end
-      end
-    end
+    result = Publishers::AtsApi::CreateVacancyService.call(permitted_vacancy_params)
+
+    render result.slice(:json, :status)
   end
-  # rubocop:enable Rails/SaveBang
 
   def update
-    @vacancy = vacancy
-
-    respond_to(&:json)
+    vacancy = Vacancy.find(params[:id])
+    result = Publishers::AtsApi::UpdateVacancyService.call(vacancy, permitted_vacancy_params)
+
+    if result[:success]
+      @vacancy = vacancy
+      render :show
+    else
+      render json: { errors: result[:errors] }, status: :unprocessable_entity
+    end
   end
 
-  def destroy; end
+  def destroy
+    vacancy = Vacancy.find(params[:id])
+    vacancy.destroy!
 
-  private
+    head :no_content
+  end
 
-  def vacancy_params
-    p = params.fetch(:vacancy).permit(:external_advert_url, :external_reference, :visa_sponsorship_available, :is_job_share,
-                                      :expires_at, :job_title, :skills_and_experience, :is_parental_leave_cover, :salary,
-                                      :job_roles, :working_patterns, :contract_type, :phases, school_urns: [])
+  private
 
-    p.except(:school_urns)
-          .merge(organisations: p.fetch(:school_urns, []).map { School.find_by(urn: _1) }.compact)
+  def required_vacancy_keys
+    %i[
+      external_advert_url
+      expires_at
+      job_title
+      skills_and_experience
+      salary
+      visa_sponsorship_available
+      external_reference
+      is_job_share
+      job_roles
+      working_patterns
+      contract_type
+      phases
+      schools
+    ]
   end
```

Comment:
This seems the ideal candidate to be turned into a `REQUIRED_VACANCY_KEYS` constant.